src/Configuration/Programmer/Programmer.cpp

```diff
@@ -183,16 +192,52 @@ void programmer_entry(const CFGCommon_ARG* cmdarg) {
           CFG_POST_ERR("Failed to list devices. Error code: %d", status);
           return;
         }
-        printDeviceList(cable, devices);
+        if (list_device->verbose) {
+          printDeviceList(cable, devices);
```

Comment:
this will brake GUI integration since there is if (Gui::GuiInterface()) Gui::GuiInterface()->Devices(cable, deviceList); call inside

Reply:
this line `if (Gui::GuiInterface()) Gui::GuiInterface()->Devices(cable, deviceList);` was added by you

What's your recommendation for the fix?

Reply:
Wrapped with the new function call `processDeviceList`